### PR TITLE
Add ThrowException action to pass exception out for string error message

### LINKED ISSUE #4119: EmitEvent action should map strings passed with "error" events to Exceptions
**Is your feature request related to a problem? Please describe.**
So in composer you can emit an "error" event but you can only pas simple values like strings or JObjects as the `value` of an event.  The "error" event is special in that it expects and `Exception` as the value.  

**Describe the solution you'd like**
To better support "error" events raised by composer flows, the `EmitEvent` action should map string values into Exception objects when an "error" event is raised.

**Additional context**

[enhancement]

### PR DESCRIPTION
**Pull request by feich-ms**
Fixes #4119

## Description
Add ThrowException action in adaptive actions. This action will wrap error of string format to Exception class and then set it into error dialog event value. OnError trigger can consume the Exception value by `turn.dialogEvent.Value`

## Specific Changes
1. add ThrowException action and its schemas
2. register it
3. add test

## Testing
Action_ThrowException is added

### CODE REVIEW

**Comment by feich-ms** [comment]
@Stevenic @vishwacsena could you help to review and verify if this PR is solving the issue #4119. Another alternative is we don't add this ThrowException action, instead we add some logic in EmitEvent action to wrap error event value to Exception type if it's not. What is your suggestion?

**Comment by boydc2014** [comment]
I personally prefer we do the trick inside EmitEvent than creating a new action just for throwing error. what do you think @Stevenic?

**Review by tomlm — APPROVED** [review]
:shipit:

**Comment by mrivera-ms** [comment]
@feich-ms , please resolve the conflicts so we can merge this PR. Thank you.

**Comment by boydc2014** [comment]
@tomlm did you think this a better fix than doing the logic inside emitEvent (which is perhaps my consistent) ?

@mrivera-ms can we onhold merge this till i get the design issues clean with @tomlm  and @Stevenic?

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/ThrowException.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        /// </summary>
+        /// <value>
+        /// A value indicating whether gets or sets whether the event should bubble to parents or not.
+        /// </value>
+        [JsonProperty("bubbleEvent")]
+        public BoolExpression BubbleEvent { get; set; }
+
+        public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (options is CancellationToken)
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/ThrowException.cs:68]: So we triaged that we will take this new action.

Can you made the implementation to be just throwing?

Comment (feich-ms) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/ThrowException.cs:73]: fixed

**Pull request merged**